azure-pipelines.yml

```diff
@@ -29,11 +29,27 @@ jobs:
   - script: python3 -m pip install pipenv && pipenv install --dev --deploy --system && python3 -m pip install flake8-formatter-junit-xml
```

Comment:
I'm all for making things nuclear bomb proof, but I'm not really convinced that the added burdeon of us evaluating new releases of pipenv to make sure they still install a project is a worthwhile use of our time. I'm comfortable hedging on the likelihood of PyPA breaking pipenv so fundamentally being quite low.

I don't mind pinning the formatter in the pipfile, it would help for local testing anyway.

Reply:
Well, I've experienced my fair share of pipenv bugs. Though I think most of them came from the use of newer pip versions which may happen if Microsoft updates the Ubuntu agent we use. They still keep up with the latest Python versions clearly since the agent currently uses 3.7.4. I don't know if they update pip when they update Python.

> I don't mind pinning the formatter in the pipfile, it would help for local testing anyway.
It's fine to pin it without putting it in the Pipfile (by specifying the version in the install command here).

Reply:
On that note we may be screwed anyway if a new Python version and new pip version are added to the agent. It doesn't look like MS keeps around old patch versions of Python, so we couldn't pin to that.

Reply:
I think this would be better hashed out at the organization level, since it would potentially affect most of our projects.

Reply:
Yeah, it sounds like the impact is a bit larger than the scope of just this project. Definitely worth discussing more in future though as if it's a serious concern, it's best to be addressed across all projects that could be impacted. I'll resolve for now, but consider making an issue ticket in org.